### ISSUE TITLE
How to use the segmentation model

### ISSUE THREAD

**Issue by Rabibabo** [matlab-deep-learning/Pretrained-YOLOv8-Network-For-Object-Detection#14]
How to initialise a matlab yolov8 segmentation model using parameters from my training. The trained parameters come from python trained code saved in onnx format

![Image](https://github.com/user-attachments/assets/646ab656-a390-4970-8469-35eaf68d7c15)

**Comment by Rabibabo** [comment]
in my_file the code like this:
[masks,labelIds, scores, bboxes] = yolov8SegPredict(img,5);
in yolov8SegPredict.m file, the code like this(I have changed it):
``` matlab
function [masks,labelIds, scores, bboxes] = yolov8SegPredict(image,numClasses)

% Copyright 2025 The MathWorks, Inc.

% Load pretrained network.
persistent net
if isempty(net)
    % net = coder.loadDeepLearningNetwork('yolov8mSeg.mat');
    net = importYOLOv8SegmentationModel("D:\CY_WORK\competition\SimSort_Competition_v2\Pretrained-YOLOv8-Network-For-Object-Detection\models\best.onnx");
end

% Get the input size of the network.
inputSize = [640 640 3];

% Apply Preprocessing on the input image.
origSize = size(image);
Ibgr = image(:,:,[3,2,1]);
img = helper.preprocess(Ibgr, inputSize);
newSize = size(img);
img = img(:,:,[3,2,1]);

% Convert to dlarray.
dlInput = dlarray(img, 'SSCB');

% Perform prediction on the input image.
outFeatureMaps = cell(7,1);
[outFeatureMaps{:}] = predict(net, dlInput);

% Apply postprocessing on the output feature maps.
[masks,labelIds, scores, bboxes] = helper.postprocessYOLOv8Seg(outFeatureMaps, ...
    origSize, newSize, numClasses);
end
```

the error is like this:
```shell
错误使用 reshape
已知维度的乘积 144 不能整除元素总数 441600。

出错 dlarray/reshape (第 40 行)
objdata = reshape(objdata, varargin{:});
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
出错 helper.postprocessYOLOv8Seg>iReshapePredictions (第 44 行)
Z1xCat = reshape(Z1ViewxCat,[],144,batchSize);
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
出错 helper.postprocessYOLOv8Seg (第 6 行)
[detectionPriors, detectionDimension, maskPriors] = iReshapePredictions(outFeatureMaps);
                                                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
出错 yolov8SegPredict (第 30 行)
[masks,labelIds, scores, bboxes] = helper.postprocessYOLOv8Seg(outFeatureMaps, ...
                                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
出错 yolo_download (第 25 行)
[masks,labelIds, scores, bboxes] = yolov8SegPredict(img,5);
```

**Comment by viakkala** [comment]
Hi,

This issue occurs when the size of 'Z1ViewxCat' is not divisible by 144. Can you validate that the input values to helper.postprocessYOLOv8Seg i.e.,  'origSize', 'newSize', and 'numClasses' are correct?

Also can you verify if the value of 'batchSize' in 出错 helper.postprocessYOLOv8Seg>iReshapePredictions (第 44 行) is correct ?

**Comment by Rabibabo** [comment]
I have created my own post-processing function based on yours, and it has solved my problem.
